src/components/modals/SettingsModal.vue

```diff
@@ -0,0 +1,195 @@
+              :value="language"
+              name="language"
+              class="SettingsModal__inputSelect SettingsModal__select__language"
+              @input="onSelectChange"
+            />
+          </ListDividedItem>
+        </ListDivided>
+      </div>
+
+      <div class="SettingsModal__footer flex flex-row justify-center md:justify-start mt-5">
+        <button :disabled="isLoading" class="pager-button mr-3" @click="emitClose">
+          {{ $t("COMMON.CANCEL") }}
+        </button>
+        <button :disabled="isLoading" class="action-button py-4 px-8" @click="save">
+          {{ $t("COMMON.SAVE") }}
+        </button>
+      </div>
+    </div>
+  </Modal>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from "vue-property-decorator";
+import { mapGetters } from "vuex";
+import moment from "moment";
+import { I18N } from "@/config";
+import CryptoCompareService from "@/services/crypto-compare";
+import { ListDivided, ListDividedItem } from "@/components/utils/listDivided";
+
+@Component({
+  components: {
+    ListDivided,
+    ListDividedItem,
+  },
+  computed: {
+    ...mapGetters("network", ["currencies"]),
+  },
+})
+export default class SettingsModal extends Vue {
+  private isLoading = false;
+  private currencies: string[];
+  private currencyName: string;
+  private currencySymbol: string;
+  private nightMode: boolean;
+  private chartMode: boolean;
+  private language: string;
+
+  get selectCurrencies() {
+    return Object.keys(this.currencies).map((currency: string) => ({
+      value: currency,
+      display: currency,
+    }));
+  }
+
+  get selectLanguages() {
+    return Object.keys(I18N.languages).map((language: string) => ({
+      value: language,
+      display: I18N.languages[language],
+    }));
+  }
+
+  public created() {
+    this.currencyName = this.$store.getters["currency/name"];
+    this.currencySymbol = this.$store.getters["currency/symbol"];
+    this.nightMode = this.$store.getters["ui/nightMode"];
+    this.chartMode = this.$store.getters["ui/priceChartOptions"].enabled;
+    this.language = this.$store.getters["ui/language"];
+  }
+
+  private onSelectChange(event: any) {
+    const { name, value } = event.target;
+
+    if (name === "currency") {
+      this.currencyName = value;
+      this.currencySymbol = this.currencies[value];
+    } else if (name === "language") {
+      this.language = value;
+    }
+  }
+
+  private storeCurrency(currency: string, rate: number, symbol: string) {
+    this.$store.dispatch("currency/setName", currency);
+    this.$store.dispatch("currency/setRate", rate);
+    this.$store.dispatch("currency/setSymbol", symbol);
+  }
+
+  private setLanguage(language: string) {
+    this.$store.dispatch("ui/setLanguage", language);
+    this.$i18n.locale = language;
+
+    this.$store.dispatch("ui/setLocale", language);
+    moment.locale(language);
+  }
+
+  private toggleTheme(enabled: boolean) {
+    this.nightMode = enabled;
+  }
```

Comment:
```suggestion
  private toggleTheme() {
    this.nightMode = !this.nightMode;
  }
```

similar for `toggleChart`, as it's not really toggling if you make it set a specific value